tests/src/core/vector/testqgsvectorlayerundocommand.cpp

```diff
@@ -0,0 +1,96 @@
+/***************************************************************************
+     testqgsvectorlayerundocommand.cpp
+     --------------------------------------
+    Date                 : May 2022
+    Copyright            : (C) 2022 by Sandro Santilli
+    Email                : strk at kbt dot io
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgstest.h"
+#include <QObject>
+#include <QString>
+
+#include "qgsvectorlayer.h"
+#include "qgsvectorlayerundocommand.h"
+
+class TestQgsVectorLayerUndoCommand: public QObject
```

Comment:
@strk can you add this to the existing testqgsvectorlayer.cpp file instead? Adding new .cpp test files should be a last resort, as they all add a significant cost to the compile time.

Reply:
I think having one test file for each QGIS class is easier to maintain (would be harder otherwise to find the test for a class). It is also easier for developers to add more test coverage for a single class: the tests I added are really only there to trigger the bug fixed by this PR but there's more to test.

If build time is an issue I suggest to file it (with the ["testsuite" label](https://github.com/qgis/QGIS/labels/testsuite)) so we can think of a way to solve that problem (there are different possible options)